src/Workspaces/Core/Portable/Workspace/Solution/TextDocument.cs

```diff
@@ -75,6 +75,16 @@ public Task<SourceText> GetTextAsync(CancellationToken cancellationToken = defau
             return State.GetTextAsync(cancellationToken);
         }
 
+        /// <summary>
+        /// Fetches the current text for the document synchronously.
+        /// </summary>
+        /// <remarks>This is internal for the same reason <see cref="Document.GetSyntaxTreeSynchronously(CancellationToken)"/> is internal:
+        /// we have specialized cases where we need it, but we worry that making it public will do more harm than good.</remarks>
+        internal SourceText GetTextSynchronously(CancellationToken cancellationToken)
```

Comment:
❓ Why not `GetText`? The vs-threading analyzers have special rules targeting cases where the synchronous form of a method is called from within an asynchronous method. If we can adopt the standard naming patterns, then these analyzers will automatically work in the expected manner if/when we adopt them.

Reply:
Following the naming pattern that was established for GetSyntaxTreeSynchronously(). At the time, we wanted to call this out explicitly in Roslyn code.

Going to favor consistency; if we want to change, that will be a separate PR.

Reply:
> Following the naming pattern that was established for GetSyntaxTreeSynchronously() ... Going to favor consistency

Well that's extremely unfortunate, but a reasonable direction for the PR at this point.